internal/nexodus/nexodus.go

```diff
@@ -873,42 +873,42 @@ func (ax *Nexodus) reconcileDeviceCache() error {
 		existing.lastHandshake = curStats.LatestHandshake
 		existing.lastRefresh = time.Now()
 		existing.endpoint = curStats.Endpoint
-		existing.peerHealthy = ax.peerIsHealthy(existing)
-		ax.deviceCache[p.PublicKey] = existing
+		existing.peerHealthy = nx.peerIsHealthy(existing)
+		nx.deviceCache[p.PublicKey] = existing
 	}
 
 	// Refresh wireguard peer configuration, getting any new peers or changes to existing peers
-	updatePeers := ax.buildPeersConfig()
+	updatePeers := nx.buildPeersConfig()
 	if newLocalConfig || len(updatePeers) > 0 {
 		// Reset connection health tracking data for any peers that have changed
 		for _, peer := range updatePeers {
-			existing, ok := ax.deviceCache[peer.PublicKey]
+			existing, ok := nx.deviceCache[peer.PublicKey]
 			if !ok {
 				continue
 			}
-			ax.logger.Debugf("resetting connection health tracking for peer %s", peer.PublicKey)
+			nx.logger.Debugf("resetting connection health tracking for peer %s", peer.PublicKey)
 			existing.startTime = time.Now()
-			if peerConfig, ok := ax.wgConfig.Peers[peer.PublicKey]; ok {
+			if peerConfig, ok := nx.wgConfig.Peers[peer.PublicKey]; ok {
 				existing.endpoint = peerConfig.Endpoint
 			}
-			ax.deviceCache[peer.PublicKey] = existing
+			nx.deviceCache[peer.PublicKey] = existing
 		}
 
 		// Deploy updated wireguard peer configuration
-		if err := ax.DeployWireguardConfig(updatePeers); err != nil {
+		if err := nx.DeployWireguardConfig(updatePeers); err != nil {
 			if strings.Contains(err.Error(), securityGroupErr.Error()) {
 				return err
 			}
 			// If the wireguard configuration fails, we should wipe out our peer list
 			// so it is rebuilt and reconfigured from scratch the next time around.
-			ax.wgConfig.Peers = nil
+			nx.wgConfig.Peers = nil
 			return err
```

Comment:
@russellb @nerdalert Would be good to get your thoughts around this, now that we have a few more places within `DeployWireguardConfig` that could fail, should we still continue to do this or should we be looking at what updatedPeers failed to be configured and only remove them from the peer list?

Reply:
@Jahmilli I think that definitely sounds more ideal. If you remove that peer from the device cache it will force an attempt to re-add it on the next peer listing.  That said, I'm not sure that would get triggered with the peer listing watch. @chirino could answer that.

If it failed on the security group, you can remove the security group ID from the peer in device cache and it would get triggered since security groups are on a poller rather than a watch currently. Sorry its not a super clean answer here. If it feels overly complex we can pop a new issue and tackle it in another PR. Thanks!!

Reply:
I would suggest leaving it as-is and considering further changes in later changes. The current behavior is the safest.  "We failed to apply the config; start over and try again."

Whether it is this change or a peer-specific change, none of that will trigger a retry. The retry will happen either when a device change comes in from watching the API, or when the reconcile timer is hit. There's still a timer that runs this code every 20 seconds, just in case.